File: Test_Coverage/processed_dataset/tests/test_HumanEval_8.py
from functions.HumanEval_8 import sum_product

def test_HumanEval_8():
     assert sum_product([]) == (0, 1)
     assert sum_product([1, 1, 1]) == (3, 1)
     assert sum_product([100, 0]) == (100, 0)
     assert sum_product([3, 5, 7]) == (3 + 5 + 7, 3 * 5 * 7)
     assert sum_product([10]) == (10, 10)

File: Test_Coverage_Java/src/test/java/test_HumanEval_8.java
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class HumanEval8Test {

    @Test
    public void testHumanEval_8_EmptyList() {
        Object[] result = sumProduct(new int[]{});
        assertEquals(0, (int)result[0]);
        assertEquals(1, (int)result[1]);
    }

    @Test
    public void testHumanEval_8_SingleElementList() {
        Object[] result = sumProduct(new int[]{10});
        assertEquals(10, (int)result[0]);
        assertEquals(10, (int)result[1]);
    }

    @Test
    public void testHumanEval_8_SameElementsList() {
        Object[] result = sumProduct(new int[]{1, 1, 1});
        assertEquals(3, (int)result[0]);
        assertEquals(1, (int)result[1]);
    }

    @Test
    public void testHumanEval_8_DifferentNumbersList() {
        Object[] result = sumProduct(new int[]{10, 5, 7});
        assertEquals(22, (int)result[0]);
        assertEquals(175, (int)result[1]);
    }

    @Test
    public void testHumanEval_8_ZeroElementList() {
        Object[] result = sumProduct(new int[]{100, 0});
        assertEquals(100, (int)result[0]);
        assertEquals(0, (int)result[1]);
    }
}